src/test/scala/uk/gov/hmrc/api/utils/BaseSpec.scala

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.api.utils
+
+import org.scalatest.featurespec.AnyFeatureSpec
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.{BeforeAndAfterAll, GivenWhenThen}
+import play.api.libs.json
+import play.api.libs.json.{JsValue, Json}
+import play.api.libs.ws.StandaloneWSResponse
+import uk.gov.hmrc.api.helpers.*
+import uk.gov.hmrc.api.service.*
+import uk.gov.hmrc.api.utils.MockMonthlyReturnData.validNdjsonTestData
+
+import java.time.LocalDate
+
+trait BaseSpec extends AnyFeatureSpec with GivenWhenThen with Matchers with BeforeAndAfterAll {
+  val authHelper: AuthHelper                                                 = new AuthHelper
+  val authToken: String                                                      = authHelper.getAuthBearerToken
+  val ppnsService: PPNSService                                               = new PPNSService
+  var clientId: String                                                       = _
+  val disaReturnsStubService: DisaReturnsStubService                         = new DisaReturnsStubService
+  val initialiseReturnsSubmissionService: InitialiseReturnsSubmissionService = new InitialiseReturnsSubmissionService
+  val monthlyReturnsSubmissionService: MonthlyReturnsSubmissionService       = new MonthlyReturnsSubmissionService
+  val completeMonthlyReturnsService: CompleteMonthlyReturns                  = new CompleteMonthlyReturns
+
+  val currentYear: Int       = LocalDate.now.getYear
+  val isaReferenceId: String = "Z4512"
+
+  override def beforeAll(): Unit = {
+    super.beforeAll()
+    val thirdPartyApplicationResponse: StandaloneWSResponse =
+      ppnsService.createClientApplication(thirdpartyApplicationHadersMap)
+    val json: JsValue                                       = Json.parse(thirdPartyApplicationResponse.body)
+    clientId = (json \ "details" \ "token" \ "clientId").as[String]
+    ppnsService.createNotificationBox(clientId, notificationBoxHadersMap).status shouldBe 201
```

Comment:
I don't love this, I'm going to look at refactoring so it logs & fails gracefully. Save any issues in the future identifying failures around test setup